backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/calendar/impl/CalendarBL.java

```diff
@@ -214,4 +227,29 @@ public IBusinessDayMatcher createBusinessDayMatcherExcluding(final Set<DayOfWeek
 				.excludeWeekendDays(excludeWeekendDays)
 				.build();
 	}
+
+	@Override
+	@NonNull
+	public CalendarId getOrgCalendarOrDefault(final @NonNull OrgId orgId)
+	{
+		final OrgInfo orgInfo = orgDAO.getOrgInfoByIdInTrx(orgId);
+		if (orgInfo.getCalendarId() != null)
+		{
+			return orgInfo.getCalendarId();
+		}
+
+		final I_C_Calendar calendar = queryBL.createQueryBuilder(I_C_Calendar.class)
+				.addEqualsFilter(I_C_Calendar.COLUMNNAME_IsDefault, true)
+				.addInArrayFilter(I_C_Calendar.COLUMNNAME_AD_Org_ID, ImmutableList.of(orgId, OrgId.ANY))
+				.orderByDescending(I_C_Calendar.COLUMNNAME_AD_Org_ID)
+				.create()
+				.first();
+
+		if (calendar == null)
+		{
+			throw new AdempiereException(MSG_SET_DEFAULT_OR_ORG_CALENDAR);
+		}
+
+		return CalendarId.ofRepoId(calendar.getC_Calendar_ID());
```

Comment:
lines 241:253 shall be a DAO method

Reply:
hehe, initially i created it in DAO, then moved it here since i believed it should not be duplicated

Reply:
done